lib/widgets/totals_table.dart

```diff
@@ -1,79 +1,92 @@
-    {
-      'label': 'DM Intake\n(kg/d)',
-      'key': 'dmIntake',
-      'decimals': 2,
-      'reqKey': 'dmIntake',
-      'isPercentage': false,
-    },
-    {
-      'label': 'ME Intake\n(MJ/d)',
-      'key': 'meIntake',
-      'decimals': 2,
-      'reqKey': 'meIntake',
-      'isPercentage': false,
-    },
-    {
-      'label': 'CP Intake\n(%)',
-      'key': 'cpIntake',
-      'decimals': 2,
-      'reqKey': 'cpIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'NDF Intake\n(%)',
-      'key': 'ndfIntake',
-      'decimals': 2,
-      'reqKey': 'ndfIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Ca Intake\n(%)',
-      'key': 'caIntake',
-      'decimals': 2,
-      'reqKey': 'caIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'P Intake\n(%)',
-      'key': 'pIntake',
-      'decimals': 2,
-      'reqKey': 'pIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Concentrate Intake\n(%)',
-      'key': 'concentrateIntake',
-      'decimals': 2,
-      'reqKey': 'concentrateIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Cost\n(ERN)',
-      'key': 'cost',
-      'decimals': 2,
-      'reqKey': null,
-      'isPercentage': false,
-    },
-  ];
+  TotalsTable(
+      {super.key,
+      required this.fodderItems,
+      required this.concentrateItems,
+      required this.cowRequirements,
+      required this.cowCharacteristics});
+
+  bool get isLactating => cowCharacteristics.lactationStage != 'Dry';
+
+  List<Map<String, dynamic>> get columns {
+    return [
+      {
+        'label': 'DM Intake\n(kg/d)',
+        'key': 'dmIntake',
+        'decimals': 2,
+        'reqKey': 'dmIntake',
+        'min': 0.95,
+        'max': 1.05
+      },
+      {
+        'label': 'ME Intake\n(MJ/d)',
+        'key': 'meIntake',
+        'decimals': 2,
+        'reqKey': 'meIntake',
+        'min': 0.95,
+        'max': 1.05
+      },
+      {
+        'label': 'CP Intake\n(%DM)',
+        'key': 'cpIntake',
+        'decimals': 2,
+        'reqKey': 'cpIntake',
+        'min': isLactating ? 16 : 12,
+        'max': isLactating ? 18 : 14,
+      },
+      {
+        'label': 'NDF Intake\n(%DM)',
+        'key': 'ndfIntake',
+        'decimals': 2,
+        'reqKey': 'ndfIntake',
+        'min': 28,
+        'max': 40
+      },
+      {
+        'label': 'Ca Intake\n(%DM)',
+        'key': 'caIntake',
+        'decimals': 2,
+        'reqKey': 'caIntake',
+        'min': isLactating ? 0.7 : 0.35,
+        'max': isLactating ? 1.0 : 0.55,
+      },
+      {
+        'label': 'P Intake\n(%DM)',
+        'key': 'pIntake',
+        'decimals': 2,
+        'reqKey': 'pIntake',
+        'min': isLactating ? 0.35 : 0.25,
+        'max': isLactating ? 0.45 : 0.40,
+      },
+      {
+        'label': 'Concentrate Intake\n(%DM)',
+        'key': 'concentrateIntake',
+        'decimals': 2,
+        'reqKey': 'concentrateIntake',
+        'min': 0,
+        'max': 60
```

Comment:
I'm not convinced that the concentrate intake is behaving. When the plan initially loads, and there is no fodder or concentrate, then the percentage displays as "NaN", which is probably too confusing for users. and should be changed. 

In the PR demo, when I click around, it seems all concentrate percentages are accepted. (I tried putting only concentrate in the feed plan, and it was still accepted. 

I think the max value should be `0.6`?

Reply:
I think the demo is from the original PR, not the updated one. Does the demo update with each new commit?

Reply:
<img width="372" alt="image" src="https://github.com/user-attachments/assets/761e3285-9660-41e3-8209-059bd92a4c60">

Reply:
The NaN doesn't show anymore. 
About the only concentrates, yeah it should be 100%. I've fixed that now.
I had done `item['concentrateIntake']` instead of `item['dmIntake']` in the loop for getting dmIntake totals for concentrate items.

<img width="372" alt="image" src="https://github.com/user-attachments/assets/02365fa2-29b4-4066-a1e1-68628eab4f13">